lib/src/client/transport/xhr_transport.dart

```diff
@@ -45,23 +46,22 @@ class XhrTransportStream implements GrpcTransportStream {
   @override
   StreamSink<List<int>> get outgoingMessages => _outgoingMessages.sink;
 
-  XhrTransportStream(this._request,
-      {required ErrorHandler onError, required onDone})
+  XhrTransportStream(this._request, {required ErrorHandler onError, required onDone})
       : _onError = onError,
         _onDone = onDone {
     _outgoingMessages.stream
         .map(frame)
-        .listen((data) => _request.send(data), cancelOnError: true);
+        .listen((data) => _request.send(Int8List.fromList(data).toJS), cancelOnError: true, onError: _onError);
 
-    _request.onReadyStateChange.listen((data) {
+    _request.onReadyStateChange.listen((_) {
       if (_incomingProcessor.isClosed) {
         return;
       }
       switch (_request.readyState) {
-        case HttpRequest.HEADERS_RECEIVED:
+        case 2:
```

Comment:
It looks like the constants are available in package:web:

https://github.com/dart-lang/web/blob/4996dc2acd30ff06f7e500ec76fde8a66db82c0c/lib/src/dom/xhr.dart#L94-L98

Reply:
I tried to write `case XMLHttpRequest.HEADERS_RECEIVED:` but i got an error from IDEA: 

> The expression of a constant pattern must be a valid constant. 
> Try making the expression a valid constant.

and compile error:

```
../grpc-dart/lib/src/client/transport/xhr_transport.dart:61:29: Error: Constant evaluation error:
        case XMLHttpRequest.HEADERS_RECEIVED:
                            ^
../grpc-dart/lib/src/client/transport/xhr_transport.dart:61:29: Context: The invocation of 'XMLHttpRequest|HEADERS_RECEIVED' is not allowed in a constant expression.
        case XMLHttpRequest.HEADERS_RECEIVED:
                            ^
../grpc-dart/lib/src/client/transport/xhr_transport.dart:61:29: Context: While analyzing:
        case XMLHttpRequest.HEADERS_RECEIVED:
                            ^
Failed to compile application.
```

Reply:
@srujzs any specific reason these are generated as external getters rather than constants? makes it impossible to use them as constants in a switch.

Reply:
The only benefit to keeping them `external` is in case the browser diverges from the IDL specification, but I expect them to rarely ever happen. This came up recently elsewhere and it's better if we emit them as Dart values to support switches. I filed https://github.com/dart-lang/web/issues/285.

Reply:
@minoic I would still use the provided values, even if not constants, and just not use a switch-case for now. You could leave a todo note pointing to that bug in dart-lang/web.

Reply:
The ability to switch over them has landed and should be in `1.1.0`.